src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonSonarDefinition.java

```diff
@@ -56,7 +55,7 @@ private void addRule(NewRepository repository, Class<?> ruleClass) {
     rule.setType(RuleType.CODE_SMELL);
     rule.setStatus(RuleStatus.valueOf(ruleAnnotation.status().toUpperCase(Locale.US)));
     rule.setHtmlDescription(ruleAnnotation.description());
-    rule.setTemplate(AnnotationUtils.getAnnotation(ruleClass, RuleTemplate.class) != null);
+    rule.setTemplate(AnnotationUtils.getAnnotation(ruleClass, org.sonar.check.Rule.class) != null);
```

Comment:
any reason for not using an import? However, minimal nice to have...

Reply:
The problem was that it took the Rule class from the org.sonar.api.server.rule.RulesDefinition package and not from org.sonar.check.Rule as it should. Even when adding the import it is not 'used' and therefore automatically deleted beacuse of the IDEs save actions.